### PR TITLE
OGG audio support

### PR DESCRIPTION
**Pull request by LucvandenBrand**
As the title says.

MP3 support should be relatively easy, as it's already in ogc-rs. OGG support will be a bit more work, but it would be great to have it due to looping support. I try to hide it all behind a generic `AudioPlayer` and `Audio` objects, as I want low-level Wii stuff as hidden as possible in this project (who knows, maybe it'll allow us to reuse this for another console in the future).

My current plan is to copy: https://github.com/devkitPro/wii-examples/blob/master/audio/oggplayer/source/oggplayer.c

### CODE REVIEW

**Comment by LucvandenBrand** [comment]
@Qqwy I'm trying to add an `oggplayer` library, which requires me to update the `build-env` stage of the Earthfile.
It seems to go well when I call `earthly +build-env`, but it seems like the `build` stage  can't find the results from the changed `build-env` stage.

Am I doing something wrong? Should I be pushing some change like you say in the readme for the other stages to see the result?

**Comment by Qqwy** [comment]
@LucvandenBrand You're hitting the limitations [outlined here](https://github.com/snappy-cobra/WiiDevEnv#re-build-base-images). For now, add the new steps to the `rust-cargo-chef` target instead. After this PR is merged I'll update the intermediate Docker image that is created from build-env (which is what is used in the later stages). This step is manual because without doing this manually from time to time Docker/Earthly will decide to rebuild everything including Devkitpro etc completely from scratch.

**Comment by LucvandenBrand** [comment]
@Qqwy Thanks for your feedback, now it's working!

I'm noticing some small audio hiccups when too much is going on (build the game, then press '1' a lot), which worries me. I think it's due to the badly performing rendering code, at least I hope and not some weird connection between input interrupts and audio.
No big worries though, we still got a lot of time. I think that, after you've given your opinion on https://github.com/snappy-cobra/WiiDevEnv/pull/54 , I'll try to do some optimization (which will be easier for me with that framework). Would be great if we could do a wii flame graph, but oh well :p .

Anyways, could you review this MR and approve if you're ok with it? Thanks in advance :)

**Comment by Qqwy** [comment]
Trying to run `earthly +build-watch` right now. It produces the following error:

```
        +build-watch *local* | error: failed to run custom build command for `ogglib v0.1.0 (/app/ogglib)`

        +build-watch *local* | Caused by:
        +build-watch *local* |   process didn't exit successfully: `/build/target/debug/build/ogglib-1578b6b32820d09d/build-script-build` (exit status: 101)
        +build-watch *local* |   --- stdout
        +build-watch *local* |   cargo:rustc-link-search=native=/opt/devkitpro/devkitPPC/powerpc-eabi/lib
        +build-watch *local* |   cargo:rustc-link-search=native=/opt/devkitpro/libogc/lib/wii
        +build-watch *local* |   cargo:rustc-link-search=native=/opt/devkitpro/portlibs/wii/lib
        +build-watch *local* |   cargo:rustc-link-search=native=/opt/devkitpro/portlibs/ppc/lib
        +build-watch *local* |   cargo:rustc-link-lib=static=c
        +build-watch *local* |   cargo:rustc-link-lib=static=sysbase
        +build-watch *local* |   cargo:rustc-link-lib=static=wiiuse
        +build-watch *local* |   cargo:rustc-link-lib=static=oggplayer
        +build-watch *local* |   cargo:rustc-link-lib=static=bte
        +build-watch *local* |   cargo:rustc-link-lib=static=ogc
        +build-watch *local* |   cargo:rustc-link-lib=static=asnd
        +build-watch *local* |   cargo:rustc-link-lib=static=aesnd
        +build-watch *local* |   cargo:rustc-link-lib=static=ogg
        +build-watch *local* |   cargo:rustc-link-lib=static=vorbisidec
        +build-watch *local* |   cargo:rerun-if-changed=wrapper.h

        +build-watch *local* |   --- stderr
        +build-watch *local* |   wrapper.h:2:10: fatal error: 'oggplayer.h' file not found
        +build-watch *local* |   thread 'main' panicked at 'Unable to generate bindings: ClangDiagnostic("wrapper.h:2:10: fatal error: 'oggplayer.h' file not found\n")', ogglib/build.rs:103:10
        +build-watch *local* |   note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

```

**Comment by LucvandenBrand** [comment]
@Qqwy 

That problem will be solved by your previous comment

> After this PR is merged I'll update the intermediate Docker image that is created from build-env (which is what is used in the later stages). This step is manual because without doing this manually from time to time Docker/Earthly will decide to rebuild everything including Devkitpro etc completely from scratch.

**Comment by LucvandenBrand** [comment]
Fixed it with some duplication for now.

**Comment by LucvandenBrand** [comment]
Ok, also simplified the code now. Realized I engineered too much.

File: app/src/audio/ogg_player.rs
Comment (Qqwy) [inline app/src/audio/ogg_player.rs]: Nitpick: Instead of a boolean to indiciate looping or not, a dedicated Enum 

```rust
pub enum LoopingMode{
  pub Infinite,
  pub OneTime,
}
```

might be nicer.

Obviously non-blocking 😊

**Review by Qqwy — APPROVED** [review]
Looks like a really nice wrapper. 

Great work writing this! After your final refactoring work, it looks really clean ❤️‍🔥 !

File: app/src/audio/ogg_player.rs
Comment (LucvandenBrand) [reply app/src/audio/ogg_player.rs]: Good suggestion, nicer than the magic bool at the end, applied it :)

**Pull request merged**
